test/extensions/problem.py

```diff
@@ -108,27 +133,56 @@ def make_id(self):
         )
 
     def forward_pass(self, sample_idx=None):
-        """Do a forward pass. Return input, output, and parameters."""
+        """Do a forward pass. Return input, output, and parameters.
+
+        Args:
+            sample_idx (int): index of the sample to select
+
+        Returns:
+            tuple[torch.nn.Tensor, torch.nn.Tensor, torch.nn.Tensor]:
+                input, output, loss, each with batch axis first
+        """
         if sample_idx is None:
             input = self.input.clone().detach()
             target = self.target.clone().detach()
         else:
-            input = self.input.clone()[sample_idx, :].unsqueeze(0).detach()
             target = self.target.clone()[sample_idx].unsqueeze(0).detach()
+            input = torch.split(
+                self.input, split_size_or_sections=1, dim=self.axis_batch
```

Comment:
- `self.input.split(...)` is shorter (not sure if it works)
- make call consistent to calls in core test suite by replacing `split_size_or_sections=1` by `1`